### PR TITLE
Issue 3 - User can edit an entry.

### PR DESCRIPTION
**Pull request by ry-zhao**
ezgif com-gif-maker (1)

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -7,23 +7,37 @@ var entries = document.querySelector('ul');
...
+    data.editing = null;
+  }
   form.reset();
+  img.setAttribute('src', 'images/placeholder-image-square.jpg');
+  entries.innerHTML = '';
+  appendEntries();
   switchViews('entries');
-  window.scroll(0, 0);
+  newEditEntry.textContent = 'New Entry';
+  entriesAnchor.className = 'margin-left-1rem';
```
Comment (codingcodymiller) [inline js/main.js]: Why is this line necessary?

File: js/main.js [hunk 2]
```diff
@@ -7,23 +7,37 @@ var entries = document.querySelector('ul');
...
+    data.editing.title = form.elements.title.value;
+    data.editing.photoUrl = form.elements['photo-url'].value;
+    data.editing.notes = form.elements.notes.value;
+    edit = false;
+    data.editing = null;
+  }
   form.reset();
+  img.setAttribute('src', 'images/placeholder-image-square.jpg');
+  entries.innerHTML = '';
+  appendEntries();
```
Comment (codingcodymiller) [inline js/main.js]: Instead of destroying all the entries and rebuilding them, you should find the DOM tree for the entry that was edited and just replace that DOM tree with a new DOM tree generated with the newly edited data.

File: js/main.js [hunk 3]
```diff
@@ -7,23 +7,37 @@ var entries = document.querySelector('ul');
...
-  var entry = {};
-  entry.title = form.elements.title.value;
-  entry.photoUrl = form.elements['photo-url'].value;
-  entry.notes = form.elements.notes.value;
-  entry.id = data.nextEntryId;
-  data.nextEntryId++;
-  data.entries.unshift(entry);
-  img.setAttribute('src', 'images/placeholder-image-square.jpg');
-  entries.prepend(loadEntry(entry));
+  if (edit === false) {
```
Comment (codingcodymiller) [inline js/main.js]: Instead of making a new variable for this, you should just check if there is a truthy value in `data.editing`. If you reset `data.editing` back to `null` after every time the form is submitted, you can always count on only needing to do editing work if there is a value in `data.editing`.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

File: js/main.js [hunk 4]
```diff
@@ -8,13 +8,24 @@ var entriesAnchor = document.querySelector('#entries-anchor');
...
-  if (edit === false) {
+  if (data.editing) {
+    data.editing.title = form.elements.title.value;
+    data.editing.photoUrl = form.elements['photo-url'].value;
+    data.editing.notes = form.elements.notes.value;
+    for (var x = 0; x < entries.children.length; x++) {
+      if (data.editing.id === parseInt(entries.children[x].getAttribute('data-entry-id'))) {
+        entries.children[x].querySelector('img').setAttribute('src', data.editing.photoUrl);
+        entries.children[x].querySelector('h2').textContent = data.editing.title;
+        entries.children[x].querySelector('p').textContent = data.editing.notes;
```
Comment (codingcodymiller) [inline js/main.js]: Instead of this, I would just generate a new DOM tree with the updated values using your `loadEntry` function, and use the `replaceWith` method to replace the DOM tree you found with the edited entry's id.

https://developer.mozilla.org/en-US/docs/Web/API/Element/replaceWith

File: js/main.js (hunk 1 above)
Comment (codingcodymiller) [reply js/main.js]: Still not sure why this line is necessary 🤔 

Does the anchor tag need to look different on different views?

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

File: js/main.js (hunk 1 above)
Comment (ry-zhao) [reply js/main.js:43]: Oops, I was under the (false) impression that we had to hide the anchor element in edit mode.

Comment (ry-zhao) [reply js/main.js]: Oh oops I thought we had to hide the anchor element in edit mode. 😅

**Review by codingcodymiller — APPROVED** [review]
